marathon_acme/tests/test_sse_protocol.py

```diff
@@ -263,3 +275,137 @@ def test_multiple_events_resets_the_event_type(self):
             ('status', 'hello'),
             ('message', 'world')
         ]))
+
+
+class FakeSseResource(Resource):
+    """
+    A leaf resource that just writes SSE headers to incoming requests and
+    stores them to a queue.
+    """
+    isLeaf = True
+
+    def __init__(self):
+        self.requests = DeferredQueue()
+
+    def render(self, request):
+        request.setResponseCode(200)
+        request.setHeader('Content-Type', 'text/event-stream')
+        request.write(b'')  # Write empty bytes to flush headers
+        self.requests.put(request)
+        return NOT_DONE_YET
+
+
+def is_sse_request():
+    return MatchesAll(
+        IsInstance(Request),
+        MatchesStructure(method=Equals(b'GET')),
+        After(
+            methodcaller('getHeader', b'accept'), Equals(b'text/event-stream'))
+    )
+
+
+def is_sse_response():
+    return MatchesAll(
+        IsInstance(_Response),
+        MatchesStructure(
+            code=Equals(200),
+            headers=HasHeader('content-type', ['text/event-stream'])
+        )
+    )
+
+
+class TestSseProtocolIntegration(object):
```

Comment:
Not sure if I should just move all the SSE tests to this style where data is written to a request?... or not bother at all?